### DISCUSSION TITLE
Vercel builds using a different environment variable

### DISCUSSION THREAD

**Discussion by sidneyramos** [vercel/community#3743] [Bug Report]
### Page to Investigate

vercel.com

### Steps to Reproduce

![Screenshot 2023-08-24 at 10 36 43 am](https://github.com/vercel/community/assets/20870623/aed1f049-4fed-4e8c-997d-1ac89011d1b8)

I have 3 separate environments in my project, dev, staging and prod. The branches are develop, staging and master respectively. Last week, we observed a bug that was happening in one preview environment (staging) where it was using the generic preview environment variable (which we set up as develop). To combat this, we set up the generic Preview environment variable as the staging variable, and added a specifically targeted variable for develop. Today, we discovered that Production is now using the generic Preview variable.

To summarise:
Initial setup:
- Preview
- Preview (staging)
- Production

Then on staging branch, it was using generic Preview, hence:
- Preview -> now contains staging value
- Preview (develop)
- Preview (staging)
- Production

But now production is using preview for some reason and i dont want to use production value on the generic preview variable.
